### PR TITLE
make the entire code base clang-15 compatible

### PR DESCRIPTION
**Pull request by fabiangreffrath**

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
This is going to be a real clusterfuck!

**Comment by rfomin** [comment]
Can we try this?
```c
typedef  void (*actionf_v)();
typedef  void (*actionf_p1)( void* );
typedef  void (*actionf_p2)( void*, void* );

typedef union
{
  actionf_v	acv;
  actionf_p1	acp1;
  actionf_p2	acp2;

} actionf_t;
```

**Comment by fabiangreffrath** [comment]
> Can we try this?
> 
> ```c
> typedef  void (*actionf_v)();
> typedef  void (*actionf_p1)( void* );
> typedef  void (*actionf_p2)( void*, void* );
> 
> typedef union
> {
>   actionf_v	acv;
>   actionf_p1	acp1;
>   actionf_p2	acp2;
> 
> } actionf_t;
> ```

I was already working on this when you suggested it. :wink:

**Review by rfomin — APPROVED** [review]
LGTM, thanks for the effort!

File: src/p_enemy.c [hunk 1]
```diff
@@ -2440,22 +2440,22 @@ void A_BabyMetal(mobj_t *mo)
   A_Chase(mo);
 }
 
-void A_OpenShotgun2(player_t *player, pspdef_t *psp)
+void A_OpenShotgun2(mobj_t *mo, player_t *player, pspdef_t *psp)
```
Comment (fabiangreffrath) [inline src/p_enemy.c:2443]: How could demos even run through with these still being wrongly declared?

**Review by rfomin — COMMENTED** [review]
If you didn't know, in C `func()` means any number of arguments, while `func(void)` means no arguments: https://stackoverflow.com/questions/693788/is-it-better-to-use-c-void-arguments-void-foovoid-or-not-void-foo

**Comment by fabiangreffrath** [comment]
Ah, right, there was a difference. Thanks for reminding me!

**Comment by fabiangreffrath** [comment]
Hm, I am not so sure anymore if this solution really requires to turn all of our PSPR action pointers into three-arguments functions. Maybe this just adds more problems that it solves... 🤔

**Comment by rfomin** [comment]
We can disable `-Wstrict-prototypes` in clang15, right? https://discourse.llvm.org/t/rfc-enabling-wstrict-prototypes-by-default-in-c/60521
We have legacy codebase and I'm not sure if that's even a good idea.

**Comment by fabiangreffrath** [comment]
Indeed! I have another local branch where I attempt to fix the build with `CC=clang-15 CFLAGS="-pedantic"` and this turns out to be even more of a clusterfuck - and once I got it to compile it led to demo desyncs (and crashes). 😩

**Pull request merged**